protocol/0028-GOVE-governance.md

```diff
@@ -221,18 +223,22 @@ A proposal to create a market contains
+The parent market must be either: 
+- in one of `proposed`, `pending`, `active`, `suspended` or `trading terminated` 
+- or `settled` state but with time since settlement less than or equal `network.liquidity.successorLaunchWindowLength` 
+- or `cancelled` (closed by governance) but with the closing time less than or equal `network.liquidity.successorLaunchWindowLength`.
 The point of setting up a market to be successor of an existing market is to
-a) allow LPs continue claim their equity-like-share (ELS) by committing liquidity to the successor market during the pending period if they wish to, and
+a) allow LPs continue claim their virtual stake / equity-like-share (ELS) by committing liquidity to the successor market during the pending period if they wish to, and
 b) allow the successor market to inherit the insurance pool of the parent market. When the successor market leaves the opening auction (moves from pending to active) the amount equal to `insurancePoolFraction x parent market insurance pool balance` is transferred to the successor market insurance pool. Once the parent market moves from "trading terminated" to "settled" state, the entire remaining insurance pool of the successor market is transferred to the successor market insurance pool.
 
+If the parent market is `proposed` or `pending` or the opening auction ends after the settlement time / cancellation time plus `network.liquidity.successorLaunchWindowLength` then the parent marketId may no longer exist in core or there may be no virtual stake to claim (copy). In that case the successor market virtual stakes are initialised as if the market has no parent (and we set the parent market field in market data to null / empty indicating no parent market). 
+
+
 Note that each market can have exactly one market as a _successor_ market.
 
 - if there already is a market (possibly pending, i.e. in opening auction, see [lifecycle spec](./0043-MKTL-market_lifecycle.md)), naming a parent market, then a subsequent proposal referencing that market is rejected.
 - if there are two proposals naming the same parent market then whichever one gets into the pending state first (i.e. passes governance vote) becomes the successor of the named parent; the other proposal is cancelled with reason "parent market no longer available".
-- if there is a successor market naming a parent market that terminates and settles or is cancelled by governance before the parent market (for whatever reason) then the parent market successor is set to empty and a different successor can be proposed.
+- if there is a successor market naming a parent market that terminates and settles or is cancelled by governance before the parent market (for whatever reason) then the parent market successor field is set to empty and a different successor can be proposed by a future market proposal.
```

Comment:
parent markets don't have a successor field. Successors have a parent market field. Is there a reason why this should be a doubly linked list?

Reply:
I assumed it will be, yes. I want to have a look at a market and see immediately whether I can use it as my parent or not.

Reply:
From data-node, or does core have to provide this?

Reply:
When a new proposal comes saying marketId `xyz` is the proposed parent core needs to do validation. 

With a single linked lists I will need to look through all the markets to see if any of them use `xyz` as parent. With a doubly linked list I just go to `xyz` and check whether its available. 

To me that sounds saner (though of course there is maybe more work to ensure consistency). 

I'll leave it up to your implementation @EVODelavega and you can come back and edit this part to match implementation. 

Data node will have to provide the double linked info because hypothetically with some obscure data retention rules the successor may be gone already from the date node so the single linked version won't do, I think.

Reply:
I agree that datanode has to offer the capability to traverse the _"family tree"_/lineage of markets, and thus requires something like a doubly linked list, and datanode will therefore be able to provide you the information required (as in: market xyz has proposal abc as a successor).

As far as the core is concerned though, markets are pretty isolated things, they aren't really aware, nor do they have access to other markets that may exist. The execution engine, which instantiates the markets is aware of which market is the parent/successor, and once a successor market was enacted, all other proposals will get rejected as per spec (several competing successor proposals can exist after all). Once a successor proposal is enacted, any future proposal to succeed its parent will just get rejected outright, and by that time, the datanode can already provide the relevant information easily.

To ensure market state doesn't depend on the state of other markets once they are enacted, I think it best that core only concerns itself with setting the parent market id field on the market. Once that market is enacted, that field is either blank or not, and becomes static.